Update Zend Engine license and doc headers referencing it

weierophinney opened the pull request:
This patch accomplishes the following:

- Updates the `Zend/LICENSE` file to the 2.01 version of the license. This primarily updates the legal entity owning the license, as Zend is now a subsidiary of Perforce.
- Updates the copyright headers that refer to the Zend Engine license to update the owner and URLs referenced.

See #7166 for the conversation thread leading to this change.

cmb69 commented on Zend/zend.c:
```diff
@@ -2,12 +2,12 @@
    | that is bundled with this package in the file LICENSE, and is        |
    | available through the world-wide-web at the following url:           |
-   | http://www.zend.com/license/2_00.txt.                                |
+   | https://www.zend.com/sites/zend/files/pdfs/zend-engine-license-2.01.txt |
```
I think the "nice" boxes need to be extended.

weierophinney replied:
I was trying to keep the patch minimal. Would like to hear from the release managers on this; @ramsey - what are your opinions?

javiereguiluz replied:
Or maybe we could remove `/pdfs/` from the URL given that the served file is not a PDF.

Could Zend devops move this file to `https://www.zend.com/sites/zend/files/zend-engine-license-2.01.txt` instead? Thanks!

weierophinney replied:
Unfortunately, no. :-/

ramsey replied:
Could you rename the file to `zend-license-2.01.txt`?

salathe reviewed (commented):
Does this mean that the "The PHP License, version 3.01" (in the `LICENSE` at the top level of this repo, and at https://www.php.net/license/3_01.txt) needs to be amended (possibly under a new version)? The PHP License, currently, states: `PHP includes the Zend Engine, freely available at <http://www.zend.com>.`

cmb69 commented on Zend/LICENSE:
```diff
@@ -1,56 +1,53 @@
+  3. The names "Zend" and "Zend Engine" must not be used to endorse or
+	 promote products derived from this software without
+	 prior permission from Zend.  For written permission, please
+	 contact:license@zend.com.   
```
or maybe `mailto:license@zend.com`?

derickr commented:
Why is Zend wanting to claim rights to any new contributions? The original Zend Engine is/was indeed their copyright, but having two different licensed parts is really no longer that useful. Is there not a way how we can disrupt this link now?

krakjoe commented:
This doesn't really feel fair. It's reasonable that perforce bought the rights to the Zend engine as it was when the deal was done, but effectively claiming ownership of any new contributions does not seem ordinary, or fair. I don't remember agreeing to it, and I don't remember anyone being given the opportunity to agree either.

At this point - to be clear, the point where the entity Zend has absolutely nothing to do with PHP development - php-src needs to be licensed under one license, and it doesn't belong to perforce, or Zend.

ramsey commented:
>  php-src needs to be licensed under one license, and it doesn't belong to perforce, or Zend.

I don't disagree with this, but copyright is a really tricky subject. Technically, every contributor owns the copyrights for the portions they've contributed, and we do not require anyone to assign their copyright to any other entity—not Zend, not the PHP Group. But it's generally understood that the contributors have granted rights to the project under the same terms as the project's license (even if there's no CLA to show that agreement). So, changing the license could be problematic.

I'm going to reach out to a lawyer friend of mine who specializes in copyright law and is knowledgable about open source licensing. I want to show him this PR to see what he thinks.

In particular, I'm concerned about the fact that some of the PHP source code is licensed under the OSI-approved PHP License 3.01, while other parts of the source code are licensed under the Zend Engine License, which is not OSI-approved. It's nearly the same license as the PHP License, but since it's not listed on the OSI's website, this might be a cause for concern. However, I doubt many people are even aware of this oddity.

I'd love to see us transition everything under the same license, so I'll see what it would take to make that leap and if we'd even be able to do it.

weierophinney commented:
> Does this mean that the "The PHP License, version 3.01" (in the `LICENSE` at the top level of this repo, and at https://www.php.net/license/3_01.txt) needs to be amended (possibly under a new version)? The PHP License, currently, states: `PHP includes the Zend Engine, freely available at <http://www.zend.com>.`

No, this can stay the way it is, though, technically, the Zend Engine is freely available here in the source code; that's where we'll point people if they come to Zend. :smile:

Thanks for the feedback on the license text - will get that updated today.

weierophinney commented:
With regards to what prompted this: in the original issue (#7166), when I reviewed, I realized that a number of URLs were invalid, and that the legal entity listed had changed hands; each could have an impact on the validity of the license, and I wanted to ensure that end users had no ambiguity around that aspect. The Perforce legal team made the following changes:

- Updated the legal entity in the header of the license (Zend went from being an LLC to an incorporated business, and is now a subsidiary of another; end users should know who that entity is _now_).
- Updated the URLs for obtaining the source code of the Zend Engine. Previously, these pointed to www.zend.com - now they point **here**.

Those are the material changes. Neither Zend nor Perforce is asserting any other rights than it had previously; this is just an update to ensure the information in the license remains accurate.

(Also, regarding the statement "At this point - to be clear, the point where the entity Zend has absolutely nothing to do with PHP development": Dmitry is still employed by Zend to work on PHP!)

weierophinney commented:
And also, FTR, I'd love to see a single license for the entirety of the PHP project; I was only doing due diligence here with the status quo.

krakjoe commented:
I didn't know Dmitry was employed by Zend still, which goes some way to
show precisely that Zend doesn't have the same hand in PHP it had when
these arrangements were made.

If we agree that it would be better to distribute PHP under one license,
then who employs Dmitry is unimportant.

How do we move from here to the place we all want to be?

I'm pretty sure it doesn't start by adding the name of a new business to
the license.

On Wednesday, 11 August 2021, weierophinney ***@***.***>
wrote:

> And also, FTR, I'd love to see a single license for the entirety of the
> PHP project; I was only doing due diligence here with the status quo.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/php/php-src/pull/7358#issuecomment-896847345>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARB52XB6EPLFJ37M55POUTT4J6JDANCNFSM5B42QVYA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

ramsey commented:
> How do we move from here to the place we all want to be?

I have reached out to a lawyer to discuss this and will report back.
